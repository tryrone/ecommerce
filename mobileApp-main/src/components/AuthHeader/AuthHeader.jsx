import React from 'react';
import { View, Image } from 'react-native';
import PropTypes from 'prop-types';

import Text from 'components/Text';
import images from 'themes/images';

import styles from './AuthHeader.styles';

function AuthHeader({
  withLogo, title, subtitle,
}) {
  return (
    <View style={styles.container}>
      {withLogo && (
        <Image source={images.duniaLogo} />
      )}
      <Text style={styles.title}>{title}</Text>
      <Text style={styles.subtitle}>{subtitle}</Text>
    </View>
  );
}

AuthHeader.propTypes = {
  withLogo: PropTypes.bool,
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
};

AuthHeader.defaultProps = {
  withLogo: false,
  subtitle: '',
};

export default AuthHeader;
