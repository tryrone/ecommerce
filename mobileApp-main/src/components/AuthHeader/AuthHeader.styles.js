import { StyleSheet } from 'react-native';

import commonStyles from 'themes/commonStyles';
import styles from 'themes/fonts';
import colors from 'themes/colors';

export default StyleSheet.create({
  container: {
    width: '100%',
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 23,
  },
  title: {
    ...commonStyles.heroTitle,
    color: colors.secondaryFont,
    textAlign: 'center',
    marginTop: 20,
  },
  subtitle: {
    marginTop: 11,
    textAlign: 'center',
    lineHeight: styles.lineHeight.medium,
    letterSpacing: styles.letterSpacing.regular,
  },
});
