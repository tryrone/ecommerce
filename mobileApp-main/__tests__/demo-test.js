import React from 'react';
import {fireEvent, render, waitFor} from '@testing-library/react-native'
import Text from '../src/components/Text/Text';
import {expect, it, jest} from '@jest/globals'
import AuthHeader from '../src/components/AuthHeader/AuthHeader';

test('AuthHeader renders correctly', () => {
  const tree = render(<AuthHeader withLogo={true} title="someOne" />).toJSON();
//   AuthHeader.propTypes.subtitle = 'cjdn';
//   expect(AuthHeader.propTypes.subtitle).toEqual('cjdn');
    console.log(tree);
    expect(1).toEqual(1);
//   expect(tree).toMatchSnapshot();
});