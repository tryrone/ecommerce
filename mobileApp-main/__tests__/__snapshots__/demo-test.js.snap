// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthHeader renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#4E4E4E",
        "fontFamily": "Roboto",
        "fontSize": 17,
        "fontWeight": "400",
      },
      null,
    ]
  }
/>
`;
